### PR TITLE
Oauth GitHub

### PR DESCRIPTION
**Pull request by avarabyeu**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/auth/personal/PersonalProjectUtils.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	 */
+	public static Project generatePersonalProject(User user) {
+		Project project = new Project();
+		project.setName(personalProjectName(user.getLogin()));
+		project.setCreationDate(Date.from(ZonedDateTime.now().toInstant()));
+
+		Project.UserConfig userConfig = new Project.UserConfig();
+		userConfig.setProjectRole(ProjectRole.PROJECT_MANAGER);
+		userConfig.setProposedRole(ProjectRole.PROJECT_MANAGER);
+		project.setUsers(ImmutableMap.<String, Project.UserConfig>builder().put(user.getId(), userConfig).build());
```
Comment (DzmitryKavalets) [inline src/main/java/com/epam/reportportal/auth/personal/PersonalProjectUtils.java]: java.util.Collections.singletonMap() ?

File: src/main/java/com/epam/reportportal/auth/integration/github/GitHubTokenServices.java [hunk 2]
```diff
@@ -29,7 +53,9 @@ public OAuth2Authentication loadAuthentication(String accessToken) throws Authen
 
 		UsernamePasswordAuthenticationToken token = new UsernamePasswordAuthenticationToken(user.getId(), "N/A",
 				AuthUtils.AS_AUTHORITIES.apply(user.getRole()));
-		OAuth2Request request = new OAuth2Request(null, null, null, true, null, null, null, null, null);
+
+		Map<String, Serializable> extensionProperties =  ImmutableMap.<String, Serializable>builder().put("upstream_token", accessToken).build();
```
Comment (DzmitryKavalets) [inline src/main/java/com/epam/reportportal/auth/integration/github/GitHubTokenServices.java]: java.util.Collections.singletonMap() ?

File: src/main/java/com/epam/reportportal/auth/integration/github/GitHubUserReplicator.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+		this.dataStorage = dataStorage;
+	}
+
+	/**
+	 * Replicates GitHub user to internal database (if does NOT exist). Creates personal project for that user
+	 *
+	 * @param accessToken Access token to access GitHub
+	 * @return Internal User representation
+	 */
+	public User replicateUser(String accessToken) {
```
Comment (DzmitryKavalets) [inline src/main/java/com/epam/reportportal/auth/integration/github/GitHubUserReplicator.java]: Any tests?

File: src/main/java/com/epam/reportportal/auth/personal/PersonalProjectUtils.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	 */
+	public static Project.Configuration defaultConfiguration() {
+		Project.Configuration defaultConfig = new Project.Configuration();
+		defaultConfig.setEntryType(EntryType.PERSONAL);
+		defaultConfig.setInterruptJobTime(InterruptionJobDelay.ONE_DAY.getValue());
+		defaultConfig.setKeepLogs(KeepLogsDelay.TWO_WEEKS.getValue());
+		defaultConfig.setKeepScreenshots(KeepScreenshotsDelay.ONE_WEEK.getValue());
+		defaultConfig.setProjectSpecific(ProjectSpecific.DEFAULT);
+		defaultConfig.setStatisticsCalculationStrategy(StatisticsCalculationStrategy.TEST_BASED);
+		defaultConfig.setExternalSystem(new ArrayList<>());
```
Comment (DzmitryKavalets) [inline src/main/java/com/epam/reportportal/auth/personal/PersonalProjectUtils.java]: java.util.Collections.emptyList()

File: build.gradle [hunk 5]
```diff
@@ -33,6 +34,7 @@ description = 'EPAM Report portal. SSO Authorization Service'
 
 repositories {
     mavenCentral()
+    mavenLocal()
```
Comment (avarabyeu) [inline build.gradle]: Agreed

File: src/main/java/com/epam/reportportal/auth/personal/PersonalProjectUtils.java (hunk 1 above)
Comment (avarabyeu) [inline src/main/java/com/epam/reportportal/auth/personal/PersonalProjectUtils.java]: Agreed

**Pull request merged**